### PR TITLE
Add Pipeline Resources for Image Transfer

### PR DESCRIPTION
**Pull request by jaysterp**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jaysterp** [comment]
@sajayantony @mnltejaswini

**Comment by NullMDR** [comment]
@jaysterp Please fix the modelvalidation error

**Comment by jaysterp** [comment]
@NullMDR ModelValidation errors fixed.

**Comment by NullMDR** [comment]
@jaysterp Please fix the error in python generation:
`
[AutoRest] ERROR (Fatal/DuplicateModelCollsion): Duplicated model name with non-identical definitions
[AutoRest]     - file:///z/work/azure-rest-api-specs/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json:1684:4 ($.definitions.SourceTriggerDescriptor)
[AutoRest]     - file:///z/work/azure-rest-api-specs/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:2541:4 ($.definitions.SourceTriggerDescriptor)
[AutoRest] 
[AutoRest] ERROR (Fatal/DuplicateModelCollsion): Duplicated model name with non-identical definitions
[AutoRest]     - file:///z/work/azure-rest-api-specs/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json:2220:4 ($.definitions.TriggerProperties)
[AutoRest]     - file:///z/work/azure-rest-api-specs/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:2132:4 ($.definitions.TriggerProperties)
`

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json [hunk 1]
```diff
@@ -15,6 +15,178 @@
...
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ExportPipelineNameParameter"
+          }
+        ],
+        "responses": {
```
Comment (majastrz) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json]: >responses [](start = 9, length = 9)

You should declare a `default` response on all operations that returns the standard error response ([https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content))

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json [hunk 2]
```diff
@@ -15,6 +15,178 @@
...
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly."
+          },
+          "202": {
+            "description": "The request was successful; the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "The export pipeline does not exist in the subscription."
```
Comment (majastrz) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json]: >subscription [](start = 70, length = 12)

Should probably say "resource group".

**Comment by majastrz** [comment]
Added some minor comments. Please take a look.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json (hunk 2 above)
Comment (jaysterp) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:140]: @majastrz this phrasing is uniform with out other resource 204 messages in this swagger. Shall we update the existing ones as well?

Comment (majastrz) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:140]: Up to you, but on 2nd thought I'm ok with current phrasing as well. It's less specific than what I proposed, but it's still accurate.

---
In reply to: [419804172](https://github.com/Azure/azure-rest-api-specs/pull/8662#discussion_r419804172) [](ancestors = 419804172)

**Comment by jaysterp** [comment]
@majastrz the comments relate to general changes that involve the entire swagger. Can we open a separate PR after this one to tackle those all together?

**Comment by majastrz** [comment]
> @majastrz the comments relate to general changes that involve the entire swagger. Can we open a separate PR after this one to tackle those all together?

Yeah, I think that will be ok.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by majastrz** [comment]
Another commit is coming, so removing signoff labels for now...

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by jaysterp** [comment]
@NullMDR can you please take a look?

**Comment by jaysterp** [comment]
@NullMDR we are ready to merge

**Pull request merged**